<!--Main-->
<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>

  <node pkg="map_server" type="map_server" name="map_server" args="$(find navigation_stage)/stage_config/maps/willow-full.pgm 0.1" respawn="false" >
    <!--param name="frame_id" value="/map" /-->
  </node>

  <node pkg="stage_ros" type="stageros" name="stageros" args="$(find navigation_stage)/stage_config/worlds/willow-multi.world" respawn="false">
    <param name="base_watchdog_timeout" value="0.2"/>
  </node>

  <!-- BEGIN ROBOT 0 -->
  <group ns="robot_0">
    <param name="tf_prefix" value="robot_0" />

    <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      <remap from="map" to="/map" />
      <param name="controller_frequency" value="10.0" />
      <rosparam file="$(find navigation_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find navigation_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find navigation_stage)/move_base_config/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find navigation_stage)/move_base_config/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find navigation_stage)/move_base_config/base_local_planner_params.yaml" command="load" />
    </node>

    <!--node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false" output="screen">
      <param name="odom_frame_id" value="robot_0/odom" />
      <param name="base_frame_id" value="robot_0/base_link" />
    </node-->
    
      <node pkg="fake_localization" type="fake_localization" name="fake_localization" clear_params="true" output="screen">
     <remap from="base_pose_ground_truth" to="robot_0/odom" />
               
     <param name="global_frame_id" value="map" />
     <param name="base_frame_id" value="robot_0/base_footprint" />
     <param name="delta_x" value="-21.67" /> 
     <param name="delta_y" value="47.12" /> 
     <param name="delta_yaw" value="0.0" />
     <remap from="scan" to="base_scan"/>
  <remap from="static_map" to="/static_map"/>

     </node>

      <node pkg="tf" type="static_transform_publisher" name="map_odom" args="47.12 21.67 0 0 0 0 /map /robot_0/odom 100" />

  </group>
  

  <!-- END ROBOT 0 -->

  <!-- BEGIN ROBOT 1 -->
  <group ns="robot_1">
    <param name="tf_prefix" value="robot_1" />

    <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      <remap from="map" to="/map" />
      <param name="controller_frequency" value="10.0" />
      <rosparam file="$(find navigation_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find navigation_stage)/move_base_config/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find navigation_stage)/move_base_config/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find navigation_stage)/move_base_config/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find navigation_stage)/move_base_config/base_local_planner_params.yaml" command="load" />
    </node>

    <!--node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false">
      <param name="odom_frame_id" value="robot_1/odom" />
      <param name="base_frame_id" value="robot_1/base_link" />
    </node-->
      
   <node pkg="fake_localization" type="fake_localization" name="fake_localization" clear_params="true" output="screen">
     <remap from="base_pose_ground_truth" to="robot_1/odom" />
     <param name="global_frame_id" value="map" />
     <param name="base_frame_id" value="robot_1/base_footprint" />
   <remap from="scan" to="base_scan"/>
  <remap from="static_map" to="/static_map"/>

   </node>
    
   <node pkg="tf" type="static_transform_publisher" name="map_odom1" args="48.12 21.67 0 0 0 0 /map /robot_1/odom 100" />
  </group>
  <!-- END ROBOT 1 -->

</launch>
